### DISCUSSION TITLE
getChainId() and getAccount().chainId returns different results when wallet is connected to chain that is not passed to wagmi chains config

### DISCUSSION THREAD

**Discussion by microHoffman** [wevm/wagmi#4030] [General]
Hello, I've observed one weird thing - I'm passing chain A + B to wagmi config as supported chains, but if my wallet is connected to chain C, the `getChainId()` returns either chain A or B, while the `getAccount().chainId` returns the actual connected chain C. Is that bug or is it intentional?

**Comment by tmm** [comment]
- `getChainId` only returns chain IDs for chains that are configured in your Wagmi Config (via [`createConfig#chains`](https://wagmi.sh/core/api/createConfig#chains)). If a connector switches to an unsupported chain, `getChainId` will return the last configured chain ID.
- `getAccount().chainId` can be any chain ID since the connector can set it to whatever it wants (e.g. open your wallet app and switch to an unconfigured chain).

**Reply by microHoffman** [reply]
Thanks for clarification, that's exactly the behavior I've spotted in my app as well. 

Do you think it's good idea to add a mention about this to docs? It might be just me, but I was assuming that the `getChainId` will be equal to the connected chain id in the wallet, which is not always the case as e.g. in the case we discussed here, which resulted in some time spent debugging an issue, so mentioning in docs could potentially save someone else some confusion. If you think it is worth mentioning in the docs, I can create a PR.

**Reply by tmm** [reply]
PR welcome! Have [this description](https://wagmi.sh/core/api/actions/getChainId#return-type) at the moment.

**Reply by microHoffman** [reply]
Here is the PR:) https://github.com/wevm/wagmi/pull/4054
